### ISSUE TITLE
Feature request: remove redundant parens rewrite

### ISSUE THREAD

**Issue by olafurpg** [olafurpg/scalafmt#553]
``` scala
for {
  a <- b
  if (a)
} yield a
```

Scalafmt:

``` scala
for {
 a <- b
if (a)
} yield a
```

I would like the input to look like this:

``` scala
for {
 a <- b
if a
} yield a
```

**Comment by olafurpg** [comment]
A good place to see how this might be implemented is to look at RedundantBraces: https://github.com/olafurpg/scalafmt/blob/786fa4230840e8e377a93fe10647628975226e7f/core/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala

**Comment by ShaneDelmore** [comment]
Looks like fun. I will take a crack at this and aim to have a problem by the weekend

**Comment by olafurpg** [comment]
Cool! The tree node you're looking to match on is called Enumerator.Guard. Here's how you can poke at the structure

```scala
> clangPrint(q"for { a <- b; if (cond) } yield a".structure)
res6: String = """
Term.ForYield(Seq(Enumerator.Generator(Pat.Var.Term(Term.Name("a")), Term.Name("b")),
                  Enumerator.Guard(Term.Name("cond"))),
              Term.Name("a"))
"""
```

**Comment by olafurpg** [comment]
The PreferCurlyFors rewrite might also give some inspiration: https://github.com/olafurpg/scalafmt/blob/master/core/src/main/scala/org/scalafmt/rewrite/PreferCurlyFors.scala

**Comment by ShaneDelmore** [comment]
PR https://github.com/olafurpg/scalafmt/pull/578 created to implement this feature.

**Comment by olafurpg** [comment]
Closed by #578
